### ISSUE TITLE
[Fixed in _B06] 1.6.0.1 betas hanging constantly on all platforms

### ISSUE THREAD

**Issue by ferdnyc** [BiglySoftware/BiglyBT#568]
Twice in a row, now, I've pulled up the remote desktop for my fileserver to find BiglyBT's UI frozen, in a state where even the window manager can't kill it, only a `kill -9 $(pidof java)` is effective. But when I manually relaunched it after killing it this time, it wasn't hung and everything functioned normally, so the hangs apparently weren't happening immediately upon launch.

In the course of my investigating, I noticed that my backup directory (I have BiglyBT doing automatic daily config backups, for various reasons) contained **two** directories for 2018-08-11, _and_ both were completely empty. This felt like a clue, and sure enough ~it was.~ it was a red herring, because the issue is much worse than I initially thought.

I pulled up the Options window in my (working-just-fine-after-forcible-restart) running instance, initiated a manual backup attempt, and... boom. Instant hard hang.

![43989679-b410d3e6-9d1c-11e8-89e3-dda3610726e6](https://user-images.githubusercontent.com/538020/43989811-1bff861c-9d1f-11e8-9e90-ee18c5c4ebd8.png)

There appears to be something amiss with ~the backup process in the latest beta.~ 1.6.0.1_B03.

#### System details
```
Java 1.8.0_181 (64 bit)
  Oracle Corporation
/usr/lib/jvm/java-1.8.0-openjdk-1.8.0.181-7.b13.fc28.x86_64/jre

SWT v4757, gtk/3.22.30
Linux v4.17.12-200.fc28.x86_64, amd64 (64 bit)
B1.6.0.1_B03/4 az3
```

**Comment by ferdnyc** [comment]
Well, crap. Maybe it's not _just_ the backups, they may just be one of the possible triggering events.

I just had B03 freeze up on me again, as I was interactively using it, even though I've disabled auto-backups.

**Comment by ferdnyc** [comment]
I've manually downgraded to 1.6.0.1_B02, and so far so good. But it's already tried to auto-install _B03 once, and I'm sure it's going to try again. I forget, is there any way to actually disable the auto-update for beta versions? This situation may be a good reason to have that as an option.

For now, if the hangs do clear up running _B02, I guess I'll just leave the beta program temporarily until _B03 can be updated or withdrawn.

**Comment by kwloafman** [comment]
Ditto on both counts.  B02 is working here, and it would be nice to be able
to skip an update.

...Thanks,
...Ken

**Comment by ferdnyc** [comment]
The 1.6.0.1_B02 .jar file download link, for anyone who needs it (just replace `BiglyBT.jar`, wherever it lives, with this file):

https://files.biglybt.com/core/BiglyBT1601-B02-signed.jar

**Comment by sklm** [comment]
Confirming constant hang-ups on Ubuntu and Windows 10. Reverted to 1.6.0.0 for now.

**Comment by ferdnyc** [comment]
@sklm So there are! I thought they might be there somewhere, I just never considered that pane. ETOOMANYOPTIONS. 

(I've turned off "Auto download updates and prompt...", but left "Check for latest version periodically" enabled. That way, when a _B04 is available, I'll be notified. That seems like the best option for the moment.)

Many thanks for the pointer, and the confirmation that this affects Windows as well. I'll update my Issue title.

**Comment by ferdnyc** [comment]
(Completely off-topic: ..._**Speaking of**_ too many options, does anyone know why there's a "Configuration Settings" section at the very bottom of Options > Files, containing a "Backup configuration files for recovery purposes" checkbox? That should be totally superseded by the entire Options _pane_ for Backup & Restore, surely?

I wonder if that Options > Files > Configuration Settings checkbox even does anything? Which isn't to say I'm brave enough to mess with it, because I'm not. :grin: )

**Comment by sklm** [comment]
( @ferdnyc Found some info [here](https://github.com/BiglySoftware/BiglyBT/issues/296#issuecomment-357260737) and [here](https://wiki.vuze.com/w/UG_Options#Files_options) )

**Comment by ferdnyc** [comment]
Ohhhhhhhhh, I see. Thanks again! 

(Hmm. Maybe I'll submit a patch to make that text clearer, and actually mention the `.bak` files rather than just saying "Backup configuration files". Because that makes it _sound_ like a complete overlap with the Backup & Restore settings.)

**Comment by parg** [comment]
Please try B04 and see if things have improved

**Comment by sklm** [comment]
Haven't experienced any hanging on B04 during the first few minutes,which is much longer than B03 was. However, it is now in endless upgrade loop, I think because B04 identifies itself as B02, at least in my installation.

**Comment by ferdnyc** [comment]
For me, so far, B04 came up, loaded some new torrents, let me start them, and even got through a "Test run auto-backup now" cycle without freezing so far. I'll see how things look in a few minutes/hours.

**Comment by sklm** [comment]
I am seeding 1600+ torrents, maybe that contributes to the problem. However, B02 is handling the same number of torrents just fine.

**Comment by ferdnyc** [comment]
Yeah, I only have 163 in All and 115 in Seeding, currently. (And only 1 is Active.) 

But, so far, 16 minutes in and it's still doing fine.

**Comment by parg** [comment]
Hopefully B04 has fixed the issue - some rework in B03 resulted in a deadlock possibility that a few people have come across

**Comment by ferdnyc** [comment]
Nope, sorry. :frowning:  I came back today to find B04 frozen again, same as @sklm . It just took longer.

**Comment by ferdnyc** [comment]
Updating to B05 now that I've restarted, don't know if that changes anything re: deadlocks.

**Comment by ferdnyc** [comment]
I guess not, because B05 froze a couple of minutes after startup.

**Comment by parg** [comment]
Just in case it works can you try cranking up a browser when it is deadlocked and entering 

  http://127.0.0.1:45100/setinfo?name=AZMSG&value=AZMSG;0;config;log-diags;;

(make sure you include the double ;; at the end of that)

If this 'page' loads then upload the debug_1.log / debug_2.log as this should contain some stack traces that might pinpoint the issue

**Comment by ferdnyc** [comment]
Sure, I'll give it a try. Back to B05, waiting on a freeze.

**Comment by ferdnyc** [comment]
Freeze achieved. Page load attempted in Firefox, and... not very enlightening, I'm afraid.

![image](https://user-images.githubusercontent.com/538020/44296826-c2fc3180-a293-11e8-9fec-f053e2ec702a.png)

**Comment by ferdnyc** [comment]
Maybe-interestingly, I have the BiglyBT Web Remote plugin installed, and I _can_ log in to that even when the GUI is frozen... however, the torrent lists all show no data. Search is operational, and even works (returns results), but attempting to view any part of my Library in the web remote is a no-go — just blank tables.

**Comment by parg** [comment]
When you load that URL into Firefox (or whatever) it causes BiglyBT to dump thread information into the debug_1.log / debug_2.log files - if you locate these (in the 'logs' folder in the BiglyBT config folder) they should contain information about the UI hang

**Comment by ferdnyc** [comment]
Yeah, here are the logs, don't know if there's anything useful. `debug_2.log` is the earlier one, the session that froze starts at:
```
[8/17/18 6:57 PM] Log File Opened for BiglyBT 1.6.0.1_B05
```
And it appears that it froze at 19:18:41. Then there's a huge time jump until 03:05:57 when I started trying to load port 45100 URLs. (I tried a few experiments of my own after the seemingly-ineffective one you provided, which is why there are JSON parser errors about 30 seconds later. Then continuing over to `debug_1.log` you'll see a bunch of logs for the web remote server, before I finally killed the client and restarted it at:
```
[8/18/18 3:23 AM] Log File Opened for BiglyBT 1.6.0.1_B05
```

[bigly_freeze_logs.zip](https://github.com/BiglySoftware/BiglyBT/files/2299237/bigly_freeze_logs.zip)

**Comment by parg** [comment]
bingo, thanks!

**Comment by ferdnyc** [comment]
I was just about to post a second, clean `debug_1.log`, from a session I started right after _deleting_ both the debug logs (which then froze) — but, sounds like no need?

**Comment by parg** [comment]
Please check out B06

**Comment by ferdnyc** [comment]
Installed & running, let's see how it goes...

**Comment by sklm** [comment]
I've been on B06 for about 3-4 hours and it seems to be handling 1600 torrents just fine. Also survived a couple of restarts. So it looks like the freezes are no more :+1:

**Comment by parg** [comment]
lovely juberly!

**Comment by ferdnyc** [comment]
Yeah, I'm at 10.5 hours of uptime now, B03-05 never would've made it anywhere near this long. Looks like we're in the clear! Thanks, @parg !
